for-demos/serve_hostname/serve_hostname.go

```diff
@@ -88,8 +90,18 @@ func main() {
 			log.Printf("HTTP request from %s", r.RemoteAddr)
 			fmt.Fprintf(w, "%s", hostname)
 		})
-		go log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
+		go func() {
+			// Run in a closure so http.ListenAndServe doesn't block
+			log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
+		}()
 	}
 
-	select {}
+	// Write to stdout after receiving SIGTERM and SIGINT to help with debugging kubernetes issue #21605
+	signals := make(chan os.Signal)
+	signal.Notify(signals, syscall.SIGTERM, syscall.SIGINT)
+	select {
+	case sig := <- signals:
+		// Keep behavior consistent with how the signal is handled by default (default is to panic)
+		log.Panicf("Terminating after receiving signal: %s.\n", sig)
```

Comment:
can't you just pull from the sig chan directly instead of selecting, since there's just a single case?